### ISSUE TITLE
Which algorithm/program should be used do pixelize the text?

### ISSUE THREAD

**Issue by realpixelcode** [beurtschipper/Depix#9]
So I tried to depixelize the given string "hello from the other side" so I typed it into Windows 10's text editor, took a screenshot and pasted it into my photo editing program. But its pixelated output is very different from the example image:

Here's my original image:
![hello-other-side-original](https://user-images.githubusercontent.com/52963327/101370371-5280e080-38a1-11eb-802c-90d55746e14d.png)
<br>
Here's my pixelated image:
![hello-other-side](https://user-images.githubusercontent.com/52963327/101370416-61679300-38a1-11eb-88c5-28a47925791d.png)
<br>
For comparison here the example image:
![](https://raw.githubusercontent.com/beurtschipper/Depix/main/docs/img/Recovering_prototype_latest.png)
<br>
Here's my output:
![output3](https://user-images.githubusercontent.com/52963327/101370714-bdcab280-38a1-11eb-890c-82776d8f7d40.png)

**Comment by beurtschipper** [comment]
Well it makes sense that the pixels are different. To match the pixelized test image you'd need to start pixelization at the exact same x and y coordinates.
I think you need to make your own search image. Try pasting the sequence from images/searchimages/debruinseq.txt in notepad, and make a screenshot of that. Use that as the search image for the s-parameter. Does that improve the result?

**Comment by beurtschipper** [comment]
For example, if I use your original text as a search image:
![search_image_internet_cut](https://user-images.githubusercontent.com/4620457/101389804-393c5c00-38c2-11eb-83e4-b32070fc0359.png)
Together with your pixels:
![cut_pixels_internet](https://user-images.githubusercontent.com/4620457/101389786-35103e80-38c2-11eb-904d-c4b6db89f69d.png)
And run: `python depix.py -s ~/Desktop/search_image_internet_cut.png -p ~/Desktop/cut_pixels_internet.png -o output_internet.png`
I get:
![output_internet](https://user-images.githubusercontent.com/4620457/101389760-2e81c700-38c2-11eb-98ac-e5168af8f0c9.png)
Does that help? :)

**Comment by realpixelcode** [comment]
Thanks for the tip! I'll try that!

**Comment by realpixelcode** [comment]
Yeah, it works using the original image as the search image – so you need to be quite sure what the pixelated image contains, right? 

Anyway, thank you 👍

![](https://user-images.githubusercontent.com/52963327/101407602-064e9400-38d3-11eb-98bb-0d2f332a0957.png)

**Comment by beurtschipper** [comment]
No that was just to show that it works. It should also work if you paste the De Bruin sequence in Notepad, make a screenshot of that, and use it as a search image.

**Comment by realpixelcode** [comment]
This is my command: ```python depix.py -p images/testimages/hello-other-side.png -s images/searchimages/debruinseq-custom.png -o output-withcustombruin.png```

This is the "hello from the other side" cipher image:
![](https://user-images.githubusercontent.com/52963327/101408707-b244af00-38d4-11eb-88ac-c31c1ed9fa32.png)

This is my screenshot of your De Bruin sequence in my Windows Notepad:
![](https://user-images.githubusercontent.com/52963327/101408590-89bcb500-38d4-11eb-82a4-64507ac25179.png)

And this is the result:
![](https://user-images.githubusercontent.com/52963327/101408566-7f022000-38d4-11eb-92ec-36f43baae79c.png)

By the way: What was the alphabet and the length to create the De Bruin sequence?

**Comment by beurtschipper** [comment]
Hmm that is interesting, I'd have to look into this. Should work but I can't quickly see why. The alphabet was [0-9] and [a-Z] of length 2.
Which tool did you use for pizelization?

**Comment by realpixelcode** [comment]
I used Corel Paint Shop Pro 2018

**Comment by beurtschipper** [comment]
Thanks. I'll look into it when I have time.

**Comment by realpixelcode** [comment]
Thank you! I really appreciate your efforts creating this software and helping me! Thanks ❤️

**Comment by RicoWeng** [comment]
niubi

**Comment by beurtschipper** [comment]
Hey @realpixelcode, can you try again with the new option `--averagetype linear`? I've added support for Gimp which uses a linear sRGB averaging, maybe your image editor does the same.

**Comment by JohnMama12** [comment]
Hey, I'm having have similar issues!
when I process the image it just adds letters inside the pixelated box.
Some boxes are sometimes even unused.
Using "--averagetype linear" just gives me an error: unrecognized arguments: linear.
How should I use those arguments?

**Comment by beurtschipper** [comment]
Did you pull the latest version?
